sinks/channel/channel_sink.go

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package channel
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/hashicorp/eventlogger"
+)
+
+// ChannelSink is a sink node which sends
+// the event to a channel
+type ChannelSink struct {
+	mu sync.Mutex
+
+	eventChan chan *eventlogger.Event
```

Comment:
If the sink is only ever written to, we can type the direction of this channel
```suggestion
	eventChan chan<- *eventlogger.Event
```

Though, I'm not sure what a write only channel is useful for. Who will read from the channel?

Reply:
I believe the caller of NewChannelSink(...) is the consumer

Reply:
I agree that it should probably be a `chan<-`

Reply:
I think, we may want some sort to timeout duration for this sink.  Yes, we
should also check for ctx->Done() in Process(...), but the caller creating the
sink should perhaps specify how long to wait for writes before returning an
error from Process(...)